src/libmaxtouch/config.c

```diff
@@ -882,49 +894,73 @@ int mxt_zero_config(struct mxt_device *mxt)
 //******************************************************************************
 /// \brief  Check the checksum for a given file
 /// \return #mxt_rc
-int mxt_checkcrc(struct mxt_device *mxt, char *filename)
+int mxt_checkcrc(struct libmaxtouch_ctx *ctx, struct mxt_device *mxt, char *filename)
 {
   int ret;
   struct mxt_config cfg = {{0}};
   uint16_t obj_idx = 0;
 
   int start_pos = INT_MAX;
-  int end_pos = 0;
+  uint16_t end_pos = 0;
 
   /* Get the mxt_config and object configurations */
-  ret = mxt_get_config_from_file(mxt, filename, &cfg);
+  ret = mxt_get_config_from_file(ctx, filename, &cfg);
   if (ret)
     goto free;
 
+  if (mxt == NULL && cfg.config_type == CONFIG_RAW) {
+    mxt_err(ctx, "RAW config format only supported with chip present.");
+    ret = MXT_ERROR_NO_DEVICE;
+    goto free;
+  }
+
   /* Find limits of CRC region */
-  for (obj_idx = 0; obj_idx < mxt->info.id->num_objects; obj_idx++) {
-    struct mxt_object *object = &mxt->info.objects[obj_idx];
-    if (is_type_used_for_crc(object->type)) {
-      int sp = mxt_get_object_address(mxt, object->type, 0);
-      if (start_pos > sp)
-        start_pos = sp;
+  struct mxt_object_config *objcfg = cfg.head;
+  if (mxt == NULL && cfg.config_type == CONFIG_XCFG) {
```

Comment:
Could you combine this with the if statement on line 911, I think it will be a bit neater